Update Home Page

AristurtleDev opened the pull request:
This is a draft PR for updating the home page of the site.  As I work through this, each commit will be available to view live on the https://mgdocs.aristurtle.net demo page.

> Note: The live demo does not build the API documentation at this time in order to speed up development.

## Current Status
- [x] Fixed hero section at top so it is full width
- [x] Fixed feature section embedded inside hero section, moved below it.
- [x] Expand feature sections to include more information
- [x] Add section for recent news
- [x] Images switch based on light vs dark theme
- [x] Update footer

## Issue Tracking
- [x] Fixed: ~~Inconsistent button design in hero compare to other buttons on the page~~
- [x] Fixed: ~~Hover effect on buttons is not implemented properly at the moment, they just fade to opaque~~
- [x] Fixed: ~~Getting Involved section does not break correctly when viewed on mobile.~~

AristurtleDev commented:
Preview as of most recent commit

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/bf287521-3e00-452c-bbb3-926d0ff00df7)

AristurtleDev commented:
Feature section update, current preview
![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/eeb83539-1c42-44f7-b046-8eebedaf4b0c)

tomspilman commented:
Looking good so far.

> Feature section update, current preview

I would say make the images here a bit smaller and keep the community one as well.  But i like the direction of it.

Be sure to check how it is looking on mobile devices as well.


> Add section for recent news

I would like to see the community site feed added back.

Before i used the Discourse API to pull the most recent posts and the top news tagged posts.

I am pretty sure this can all be done in JS.

ALL THAT SAID.

We will be discussing the future of the community site this week, so not sure how that will go.

AristurtleDev commented:
@tomspilman Sorry, had just pushed a new commit right before I saw your message

> I would say make the images here a bit smaller and keep the community one as well. But i like the direction of it.

Can resize them. Wasn't too happy originally with the size either, was going to come back to it.

> Keep the community one as well

I moved the community one down to a **Getting Involved** section (see recent preview below).  I can move this back into the other section if wanted.

> Be sure to check how it is looking on mobile devices

Yep, doing mobile first design, checking against Mobile (iPhone 12/13 profile at 390x844), tables ( iPad profile at 810x1080), standard desktop (1280x720) and high resolution ( 3840x1600)

> I would like to see the community site feed added back.

Can do that, shouldn't be an issue

Current Preview after adding Getting Involved section 
![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/ed16ca0c-ce64-4652-8fb4-0f9928a64cdc)

AristurtleDev commented:
Images made smaller in features section, changed to 250px width.  Moved out of bootstrap `card` so that decreasing image didn't leave blank space in card image top section


<details>
<summary><h2>Click to view preview</h2></summary>

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/0993e685-a814-4b88-9866-3f75f7d4d66f)

</details>

AristurtleDev commented:
@tomspilman I attempted to add the latest feed from the community site, which would be at https://communiyt.monogame.net/latest.json

However, I'm unable to actually get a response through code due to CORS error.  The API expects a `Api-Key` and `Api-Username` value to successfully make the request https://docs.discourse.org/#tag/Topics/operation/listLatestTopics

Previously I believe the MonoGame site was on wordpress correct?  I imagine it used the Discourse WP plugin, which has to be configured with the key and username to use.  

If I were to put the key and username into the script here, since this is done in docfx as a static site generator, that means the API key and username would be visible in the browser from the developer tools panel.

tomspilman commented:
> If I were to put the key and username into the script here, since this is done in docfx as a static site generator, that means the API key and username would be visible in the browser from the developer tools panel.

Hum.  We're not really trying to pull secret data from the Discourse site.  I will have to investigate it a bit on that.

I know we can pull RSS data from the site freely:

https://community.monogame.net/latest.rss

Maybe that is another path if it can give us what we need.

tomspilman commented:
Oh...

![image](https://github.com/MonoGame/monogame.github.io/assets/103522/c2899bac-4061-41cd-9107-a4bfb3af49ef)

I wonder if fixing that would help here?

But it seems hard to test otherwise.

(I already removed w3schools...  not sure why that was there... someone was testing things before i guess)

AristurtleDev commented:
@tomspilman  you should be able to generate a read-only key that can be used publically https://meta.discourse.org/t/create-and-configure-an-api-key/230124

I'm using test data right now to build the feed section on the site, but as soon as i can make the request, I can get the data live
![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/36ca5a11-f42c-4b19-a646-b3d8c22f32e0)

AristurtleDev commented:
> Hum. We're not really trying to pull secret data from the Discourse site. I will have to investigate it a bit on that.

It's not so much that we'd be pulling secret data, just need to ensure that the key is a read-only scope so that nefarious users can't use it to do things like delete topics using the API

tomspilman commented:
@AristurtleDev - Ok here is the read-only API key:  `e6f61661d8e471a445a5196a4cb428c9a5adf065e1957fa984bf4431927f5bb6`

AristurtleDev commented:
@tomspilman need the username for the key as well
![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/f00f4361-9ba0-4f6b-9432-9dc21b3d0ee1)

tomspilman commented:
@AristurtleDev - I set the key as "All users" so it shouldn't need a user name?

Also added `mgdocs.aristurtle.net` as a CORS origin if that helps.

tomspilman commented:
I guess you can use `mgbot` as the user as the "All users" works for any user.

AristurtleDev commented:
@tomspilman doesn't seem to be doing it.

I added the script to mgdocs.aristurtle.net
```js
async function latestNews() {
    let headers = new Headers({
        'Api-Key': 'e6f61661d8e471a445a5196a4cb428c9a5adf065e1957fa984bf4431927f5bb6',
	'Api-Username': 'mgbot'
    });

    let init = { method: 'GET', headers: headers };
    let response = await fetch('https://community.monogame.net', init);
    let json = await response.json();
    console.log(json);
}


latestNews();
```

still get cors blocks

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/35e7463e-526a-464b-823a-3467cf9d9dbf)

AristurtleDev commented:
Should also note that the same issue occurs when attempting to `fetch` the `latest.rss` as well

tomspilman commented:
@AristurtleDev - somehow https://mgdocs.aristurtle.net/ wasn't in the CORS whitelist.  I just fixed that.  Give it another try.

AristurtleDev commented:
@tomspilman it's still blocking 😅
I may have a workaround solution for now that I can implement, but it isn't something that should be used permanently

tomspilman commented:
@AristurtleDev - Here is another key that is user specific to mgbot:  `ad7918672dc951ef177ed85bffb724b3e6df22ec7c1b218d2f05717e4ed5b6b2`

Maybe this solves things.

AristurtleDev commented:
@tomspilman unfortunately didn't fix it.  I wouldn't stress it for now, We'll get it figure out somehow, in the meantime I have a small workaround can use.

the Latest News section will roughly look like this now, still tweaking a little
If, when the team makes a news post, they include an image in the post, I can include that image in the post on the home page like below.

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/49661fe3-ac0e-49e5-a9ad-82c8aee53ac0)

tomspilman commented:
> the Latest News section will roughly look like this now, still tweaking a little

That works for me.

We can consider later if we want to show some of the general community activity like we did before as well.

AristurtleDev commented:
@tomspilman I reworked it a little, and used the foundation of it to also add the community posts.  Did that before i saw your message about considering it later.  I can remove the community posts if necessary.

The biggest thing I don't like is that discourse api doesn't always include an `excerpt` field value to pull the preview text from for the post. So some will have the preview text and others have none.

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/50224360-ef91-4e74-9463-a4091c4a7d6b)

<details><summary>Click to view full page preview</summary>

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/20771bcb-852b-439b-a814-5b97af486c46)

</details>

tomspilman commented:
A few small notes and personal opinions:

![image](https://github.com/MonoGame/monogame.github.io/assets/103522/cf51779a-e24c-4720-b4d9-7018c040e50a)

Feels like a bunch of whitespace between the features and the "get involved" and not much whitespace between the feature images and the text.

![image](https://github.com/MonoGame/monogame.github.io/assets/103522/16d8e504-53d7-4c90-8f13-e0a131a9c3ce)

Same sort of thing here....  lots of space between the headers "Latest News" and "Latest Community Posts" and the content below.  But not much space between the bottom of the news posts and the header for community posts.

Another thing to consider is dropping the boxes around the posts in favor of just good whitespace balance.

AristurtleDev commented:
@tomspilman this is what the posts section looks like with border removed. 

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/63d6d102-f05a-4218-aa34-5b4057db63f2)

I feel like without the boarders to visually separate elements, the statistic values for each post being at the bottom with no visual separator between them and the next post looks cluttered.

I can remove the statistic values and just keep the "last updated" value and move it to below the post header like this

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/b1db752d-824a-412c-8e8c-a9e1b098d74a)


Also, what would be your thoughts on combining the latest news and latest community post into one section for desktop view, side-by-side like the following

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/fa70e92e-a99c-4e21-87b2-331034b80e2f)

tomspilman commented:
> I feel like without the boarders to visually separate elements, the statistic values for each post being at the bottom with no visual separator between them and the next post looks cluttered.

Yeah to do that you would have to tighten them up to the title/content:

![image](https://github.com/MonoGame/monogame.github.io/assets/103522/97ae589c-0d17-46ce-b28d-1d8c0ba52e44)

So it feels like it is together and not part of the next thing.


> Also, what would be your thoughts on combining the latest news and latest community post into one section 

Yeah that makes sense to me.

AristurtleDev commented:
Latest News / Community section update.
- statistics line was given right padding on desktop to break it up from posts next to it
- statistic line moved up for posts that do not have excerpt
- values in statistics are shorthanded so long values are not used which cause word-wrap issues
- news and community listed side by side on desktop, individually on mobile

** Desktop View**
![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/09a1eb4c-2b95-48b9-8ec1-0023d959f7b9)

**Mobile View**
![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/5ccac5c2-3424-47ee-813d-be71c1decf90)

<details><summary>Click to view full page preview</summary>

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/420e7b5a-23d6-4882-9230-dee186154365)


</details>

tomspilman commented:
Looks pretty good to me @AristurtleDev

AristurtleDev commented:
@tomspilman At this point I think it would be ok for review and merge if approved.  There are more changes I want to add such as a featured games section as well as a sponsors section, but I have blocking info that I need before I can implement them. For instance, I've reached out to Re-Logic about what banner/logo/wordmark they would like used to show them as a sponsor and I'm waiting to get the response back.

In the mean time, we could push the current changes here and I can do separate PRs later for the other sections when I get the info needed. This would give the home page a nice update for now with more added in the future.

Live preview can be viewed live at [https://mgdocs.aristurtle.net](https://mgdocs.aristurtle.net)

<details><summary>Click to view static image preview</summary>

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/a8e1d8d9-5f1b-42f9-bc16-7b33c85a1efd)


</details>

tomspilman commented on templates/monogame/layout/_master.tmpl:
```diff
@@ -104,67 +104,118 @@
+            {{_appFooter}} <br/>
+            {{#_overrideDocFx}}
+            Home page designed with ❤ by <a href="https://github.com/AristurtleDev">Christopher Whitley (AristurlteDev)</a> with feedback from community members<br/>
+            Code, Documentation, and Report Issues icons created by community member TankStory
```
Not trying to properly attribute you on this work @AristurtleDev, but can we find a simpler way to attribute people than this?

If 2 or 3 more people start contributing do we then list all of them?  My concern is the footer getting huge with lots of credits if we set this precedent of listing people and specifically what they contributed.

Maybe we do something much simpler like this:

Designed with ❤ by [MonoGame community](https://github.com/MonoGame/monogame.github.io/graphs/contributors).

The link there being to the contributor graph for the website.  This then has everyone listed that contributed to the website with specifically how much they contributed to it.

We can then use this across all the footers on the website.

tomspilman replied:
Also wouldn't mind:

Designed with ❤ by (AristurtleDev)[https://github.com/AristurtleDev] and the [MonoGame Community](https://github.com/MonoGame/monogame.github.io/graphs/contributors).

Just want to keep it from getting much bigger over time.

AristurtleDev replied:
I like the MonoGame Community idea. I felt weird crediting myself, but it was a common thing to do, but just dropping it to community with link to the contributors is a great idea. Updated to this in https://github.com/MonoGame/monogame.github.io/pull/19/commits/d24ce652460368874111abeb1291cb179ac2ba17

tomspilman commented on index.md:
```diff
@@ -1,57 +1,198 @@
+	</div>
+	<div class="hero-credits">
+		<a href=""><img id="credits-logo" src=""></a><br/>
+		<a id="credits-built-with" href="https://twitter.com/search?q=BuiltWithMonoGame">#BuiltWithMonoGame</a>
```
This #BuiltWithMonoGame looks really hard to read.

That said... since Twitter is dying are hashtags a thing we care about anymore?

AristurtleDev replied:
I adjusted it to bold and have a text shadow to give it more emphasis.   https://github.com/MonoGame/monogame.github.io/pull/19/commits/0190c2fbee945aba8a7a89b881ced4ac4cece7d8

It more noticeable but depending on the background image in that section the colors could still blend like with Stardew Valley and Ty

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/419793a6-f097-48b2-8111-0c2f7c319429)

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/1d2bfca3-0d3e-4c1c-91c9-3214fdb49800)


Another option would be to adjust it to white text.  All links on the page are set to orange color, which is why this picks up that color value, but setting this one specifically to white make sense to make it more visible no matter the background image

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/ed1faf1e-2f03-4e7f-a9d2-cba815140395)

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/21ca31ae-7d35-4bad-9f7d-075ee8ea61ce)

tomspilman commented on index.md:
```diff
@@ -1,57 +1,198 @@
+			</div>
+		</div>
+		<div class="col-12 col-md-6">
+			<h3 class="fw-semibold mb-4 text-body-emphasis">Cross-Platform</h3>
```
There isn't much of a gap between these two sections when on mobile:

![image](https://github.com/MonoGame/monogame.github.io/assets/103522/93090361-6fec-4ced-a3e7-085b57982b53)

AristurtleDev replied:
Bottom margin (`md-5`) added after icons https://github.com/MonoGame/monogame.github.io/pull/19/commits/e92a802a2f95790cfec454145b2fe0697fbb5d65

tomspilman commented on index.md:
```diff
@@ -1,57 +1,198 @@
+		   <ul class="list-unstyled">
+			<li class="mb-3 lead"><i class="bi bi-check-lg color-monogame pe-2"></i> No licensing cost!</li>
+			<li class="mb-3 lead"><i class="bi bi-check-lg color-monogame pe-2"></i> No royalties!</li>
+			<li class="mb-3 lead"><i class="bi bi-check-lg color-monogame pe-2"></i> No runtime fee!</li>
```
Hah... love it!

AristurtleDev replied:
Haha, glad you liked this 😅

tomspilman commented on index.md:
```diff
@@ -1,57 +1,198 @@
+		   </ul>
+		</div>
+		<div class="col-12 col-md-6 d-flex flex-row justify-content-center">
+			<img id="github-logo" class="featurette-image object-fit-contain" loading="lazy">
```
My first impression is some of these icons around these sections seem overly large:

![image](https://github.com/MonoGame/monogame.github.io/assets/103522/56612432-b4f2-4b1e-b250-2a7a501de5c9)

AristurtleDev replied:
Adjusted the size of the GitHub logo image here.  Also updated the managed-code image to fix the tilt perspective on the game window overlaying the code window (they were misaligned).  

I didn't feel that the managed-code image was too large, but I'll defer to your recommendation for it

tomspilman commented on index.md:
```diff
@@ -1,57 +1,198 @@
+    <div class="row align-items-md-center g-5 py-5">
+      <div class="col">
+        <div class="row g-4">
+          <div class="col-12 col-md-4 d-flex flex-column gap-2 mb-5">
```
It may be nice if the images in these sections centered instead of left:

![image](https://github.com/MonoGame/monogame.github.io/assets/103522/03f8a988-f1da-4942-b184-a4b8d544c224)

Especially in the mobile layouts.

AristurtleDev replied:
These are now center aligned using `align-self: center`.  If the parent flex container used `align-content: center` it would center the icons and all text. https://github.com/MonoGame/monogame.github.io/pull/19/commits/c98c5706073753289af01ba67df7c7aecf694c6b

tomspilman commented on index.md:
```diff
@@ -1,57 +1,198 @@
-</div>
+<section class="hero" >
+	<div id="hero-background-image" />
+	<div class="hero-content">
```
So the darkening of the game hero image works well in dark mode:

![image](https://github.com/MonoGame/monogame.github.io/assets/103522/987f1261-9b51-497f-aee4-8491dbafaff4)

In light mode it lightens the image:

![image](https://github.com/MonoGame/monogame.github.io/assets/103522/084a9800-d5a0-423d-ad17-c4c8ef875725)

It seems to me like it should always darken even in the light mode for this particular case.

AristurtleDev replied:
This was actually supposed to be disabled and overlooked on my part. I initially added the opacity because the hero text was hard to read on some images.  I eventually switched it to use a text shadow to pop it out more, so the opacity can be removed.  

One thing I did notice however when removing the opacity, in relation to the other comment about the hash tag, it becomes even harder to read even though I've added the text shadow to it and bold styling.

![image](https://github.com/MonoGame/monogame.github.io/assets/103014489/314442b6-318f-42a6-8686-8821974f8b1b)


I'll add it here, but going to change that text to white so there are no issues with it blending with the background of any images now or added in the future to the heading section.

tomspilman commented:
Ok @AristurtleDev i added a bunch of feedback.  Almost all of it is small issues.  In the end I'm good with it being your call on this as you did the heavy work. 

> There are more changes I want to add such as a featured games section

It is starting to feel like we have everything on the home page and it is getting sort of long.  Like the news is at the bottom of the page and i expect 99% of people will never scroll that far.

To me... the hero banner is the "featured games section" of the home page.   I don't feel we need yet another place to put featured games.  If anything we could maybe improve the hero banner with a better transition, maybe navigation links to scroll thru them, etc.

> as well as a sponsors section

I would maybe put someone like Re-logic in the footer as a major sponsor.  

But unless we get more sponsors i don't know if we need a section on the home page for this.

I had this thing i made for the old PayPal sponsors under Your Donations:

https://www.monogame.rocks/donate

It may be good to honor the old sponsors somehow on the site... but maybe not with a section on the home page.

Maybe we need a Sponsors page we link from the footer or something and have old sponsors listed, information on how to sponsor us, and what you get for that sponsorship?

AristurtleDev commented:
@tomspilman went through and resolved some of the review comments. There are some where I left some feedback for additional input on changes instead of clicking "resolve conversation"

If you're good with the updates now, I'm good with finalizing this.


> To me... the hero banner is the "featured games section" of the home page. I don't feel we need yet another place to put featured games. If anything we could maybe improve the hero banner with a better transition, maybe navigation links to scroll thru them, etc.

I can work on this in a separate PR so it's not blocking this one from being pushed.  Just something simple like `<` and `>` arrows on the edges of the banner to manually click through if a viewer wants? Like a typical carousel element?



> I would maybe put someone like Re-logic in the footer as a major sponsor.

This is another I can add in a separate PR so it's not blocking the update from being pushed.  I reached out to Yoria Omer of Re-Logic to get the wordmark/logo/image they would like represented for them.  They forwarded it to the right people, just waiting to hear back from them on what to officially use as their branding.

> But unless we get more sponsors i don't know if we need a section on the home page for this.
> 
> I had this thing i made for the old PayPal sponsors under Your Donations:

Not to sound like a broken record, but this is another one I can do in a separate PR. 

Along with the original donations page, there are other pages missing from the new site, like the Downloads page.  [(Someone already mentioned it in the forums](https://community.monogame.net/t/feel-like-im-going-crazy-but-cant-find-downloads/19830/2))  I'm not sure if a Download page would need to be added back in to prevent confusion, but it's also always nice to have a page where users can select older version downloads if they need to for whatever reason.

tomspilman commented:
> If you're good with the updates now, I'm good with finalizing this.

Done... merging.

tomspilman commented on scripts/latest-posts.js:
```diff
@@ -0,0 +1,162 @@
+ * it into the #latest-community element on the page
+ */
+async function get_latest_community() {
+    let response = await fetch('https://communityapi.aristurtle.net/latest-community');
```
@AristurtleDev was it your intent to leave this linked to your own site?

Maybe now that this is published to the official MG site the direct community API link will work?
